### PR TITLE
tests: coverage increase

### LINKED ISSUE #59: tests: increase test coverage

### PR DESCRIPTION
**Pull request by omelkonian**
* Increases test coverage. (closes #59)

Signed-off-by: Orestis Melkonian <melkon.or@gmail.com>

### CODE REVIEW

File: invenio_oauthclient/models.py [hunk 1]
```diff
@@ -146,10 +146,10 @@ class RemoteToken(db.Model):
     secret = db.Column(db.Text(), default='', nullable=False)
     """Used only by OAuth 1."""
 
-    def __repr__(self):
+    def __repr__(self):  # pragma no cover
```
Comment (jirikuncar) [inline invenio_oauthclient/models.py]: Adding pragma no cover is not really the way to increase test coverage.

```python
def test_token_repr():
    """..."""
    assert 'Remote Token ...' == RemoteToken(...)
```

Comment (omelkonian) [inline invenio_oauthclient/models.py]: Sorry, assumed it was good practice as I saw it as an example on the documentation of coverage.py (http://coverage.readthedocs.io/en/coverage-4.0.3/excluding.html). Anyway, updated.

File: tests/test_models.py [hunk 2]
```diff
@@ -90,3 +90,17 @@ def test_get_regression(models_fixture):
...
+    datastore = models_fixture.extensions['invenio-accounts'].datastore
+    user = datastore.find_user(email="existing@inveniosoftware.org")
+
+    assert 'Remote Token <token_type=type access_token=mytoken>' == \
+           RemoteToken.create(user.id, "dev",
+                              "mytoken", "mysecret",
+                              token_type='type').__repr__()
+
+    assert 'Remote Account <id=1, user_id=1>' == \
+           RemoteAccount.get(user.id, "dev").__repr__()
```
Comment (jirikuncar) [inline tests/test_models.py]: `repr(RemoteAccount.get(user.id, "dev"))`

File: tests/test_models.py [hunk 3]
```diff
@@ -90,3 +90,17 @@ def test_get_regression(models_fixture):
...
+
+def test_repr(models_fixture):
+    """Test representation of RemoteAccount adn RemoteToken."""
+    datastore = models_fixture.extensions['invenio-accounts'].datastore
+    user = datastore.find_user(email="existing@inveniosoftware.org")
+
+    assert 'Remote Token <token_type=type access_token=mytoken>' == \
+           RemoteToken.create(user.id, "dev",
+                              "mytoken", "mysecret",
+                              token_type='type').__repr__()
```
Comment (jirikuncar) [inline tests/test_models.py]: use function `repr` instead of `__repr__` method

File: invenio_oauthclient/ext.py [hunk 4]
```diff
@@ -94,7 +94,7 @@ def __init__(self, app):
             )
 
             # Register sign-up handlers
-            def dummy_handler(remote, *args, **kargs):
+            def dummy_handler(remote, *args, **kargs):  # pragma no cover
```
Comment (jirikuncar) [inline invenio_oauthclient/ext.py]: IMHO it would be better to create simple test that registers new client without any handler.

**Comment by jirikuncar** [comment]
@omelkonian Good job❗  Can you please have a look at my last comments and let me know when you update the PR? Thanks

**Comment by omelkonian** [comment]
@jirikuncar LGTY?

**Comment by jirikuncar** [comment]
@omelkonian This branch is out-of-date with the base branch.

**Comment by jirikuncar** [comment]
@omelkonian please remove the merge commit.

**Comment by omelkonian** [comment]
@jirikuncar i think its good to go

**Comment by jirikuncar** [comment]
@omelkonian see http://invenio.readthedocs.io/en/latest/technology/git.html#r2-remarks-on-commit-log-messages (`tests: coverage increase`)

**Comment by jirikuncar** [comment]
@omelkonian can you please rebase?

File: tests/conftest.py [hunk 5]
```diff
@@ -182,6 +182,22 @@ def params(x):
...
 @pytest.fixture
+def remote():
+    """Fixture for remote app."""
+    return type("test_remote", (),
+                dict(
+                    name='example_remote',
+                    request_token_params={'scope': ''},
+                    base_url='https://foo.bar/',
+                    request_token_url=None,
+                    access_token_url="https://foo.bar/oauth/access_token",
```
Comment (jirikuncar) [inline tests/conftest.py]: Can you please use consistently single quotes (`'`)?

File: tests/test_contrib_github.py [hunk 6]
```diff
@@ -295,3 +298,36 @@ def login():
...
             assert RemoteAccount.query.filter_by(user_id=u.id).count() == 0
             assert RemoteToken.query.count() == 0
+
+
+def test_not_authenticated(app):
+    """Test disconnect when user is not authenticated."""
+    with app.test_client() as client:
+        assert not current_user.is_authenticated
+        resp = client.get(
+            url_for("invenio_oauthclient.disconnect", remote_app='github'))
```
Comment (jirikuncar) [inline tests/test_contrib_github.py]: https://github.com/inveniosoftware/invenio-oauthclient/pull/67/files#r72631719

File: tests/test_contrib_orcid.py [hunk 7]
```diff
@@ -247,3 +248,13 @@ def login():
...
         ).count()
+
+
+def test_not_authenticated(models_fixture):
+    """Test disconnect when user is not authenticated."""
+    app = models_fixture
+    with app.test_client() as client:
+        assert not current_user.is_authenticated
+        resp = client.get(
+            url_for("invenio_oauthclient.disconnect", remote_app='orcid'))
```
Comment (jirikuncar) [inline tests/test_contrib_orcid.py]: https://github.com/inveniosoftware/invenio-oauthclient/pull/67/files#r72631719

File: tests/test_handlers.py [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+from invenio_oauthclient.views.client import blueprint as blueprint_client
+from invenio_oauthclient.views.settings import blueprint as blueprint_settings
+
+
+def test_token_setter(app, remote):
+    """Test token setter on response from OAuth server."""
+
+    # OAuth1
+    resp_oauth1 = {
+        "name": "Josiah Carberry",
```
Comment (jirikuncar) [inline tests/test_handlers.py]: https://github.com/inveniosoftware/invenio-oauthclient/pull/67/files#r72631719

File: tests/test_handlers.py [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+    base_app.config['OAUTHCLIENT_REMOTE_APPS']['cern']['signup_handler'] = {}
+
+    # Initialize InvenioOAuth
+    FlaskOAuth(base_app)
+    InvenioOAuthClient(base_app)
+    base_app.register_blueprint(blueprint_client)
+    base_app.register_blueprint(blueprint_settings)
+
+    # Try to sign-up client
+    base_app.test_client().get(url_for("invenio_oauthclient.signup",
```
Comment (jirikuncar) [inline tests/test_handlers.py]: https://github.com/inveniosoftware/invenio-oauthclient/pull/67/files#r72631719

File: tests/test_models.py [hunk 10]
```diff
@@ -90,3 +90,17 @@ def test_get_regression(models_fixture):
         t3.remote_account.user_id
     assert RemoteToken.get(user3.id, "dev").remote_account.user_id == \
         t4.remote_account.user_id
+
+
+def test_repr(models_fixture):
+    """Test representation of RemoteAccount adn RemoteToken."""
+    datastore = models_fixture.extensions['invenio-accounts'].datastore
+    user = datastore.find_user(email="existing@inveniosoftware.org")
```
Comment (jirikuncar) [inline tests/test_models.py]: https://github.com/inveniosoftware/invenio-oauthclient/pull/67/files#r72631719

File: tests/test_views.py [hunk 11]
```diff
@@ -305,6 +305,11 @@ def test_no_remote_app(views_fixture):
         )
         assert resp.status_code == 404
 
+        resp = client.get(
+            url_for("invenio_oauthclient.signup", remote_app='invalid')
```
Comment (jirikuncar) [inline tests/test_views.py]: https://github.com/inveniosoftware/invenio-oauthclient/pull/67/files#r72631719

**Comment by omelkonian** [comment]
Is it ok that I replaced string quotes to more places than I changed?

**Comment by jirikuncar** [comment]
@omelkonian normally it would be better to do replacement in separate commit and then include your chances.

**Comment by omelkonian** [comment]
@jirikuncar I put it on a separate commit

**Pull request merged**